azure-simple/Packer/ansible-apim/roles/apim/tasks/main.yml

```diff
@@ -0,0 +1,191 @@
+#----------------------------------------------------------------------------
+#  Copyright (c) 2018 WSO2, Inc. http://www.wso2.org
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#----------------------------------------------------------------------------
+---
+  - block:
+    # Set installation directory information
+    - name: Installation Information
+      set_fact:
+        product_dir: "{{ target }}/apim"
+        carbon_home: "{{ target }}/apim/{{ product_name }}-{{ product_version }}"
+        server_backup_dir: "{{ backup_dir }}/apim"
+        shared_dir: "{{ target }}/sharedfs"
+
+    - name: Generate the random number
+      set_fact:
+        random_number: "{{ (30) | random }}"
+      run_once: yes
+
+    - name: Wait for "{{ random_number }}"s before parallel execution
+      wait_for:
+        timeout: "{{ random_number }}"
+      delegate_to: localhost
+
+    #WSO2 Distribution
+    - name: Create product directory
+      file:
+        path: "{{ product_dir }}"
+        state: directory
+        mode: 0755
+
+    - name: Copy WSO2AM service file
+      template:
+        src: wso2apim.service.j2
+        dest: /etc/systemd/system/wso2apim.service
+
+    - name: Stop wso2apim as a service
+      systemd:
+        name: wso2apim
+        state: stopped
+
+    - name: Check if deployment resource directory exists
```

Comment:
@rasanjaya85 just out of curiosity. Are we sharing the entire `<AM_HOME>/repository/deployment/server` directory in this case? As per the team's recommendation, we need to share only the specific directories such as, `executionplans`, `synapse-configs` and `userstores` (optional according to the product team and may not be required in the default installation artifacts).

Reply:
No, This release validates the  ""{{ carbon_home }}/repository/deployment/server" is exist then synchronize only the "{{ carbon_home }}/repository/deployment/server/synapse-configs" files to mounted data disk. This behavior has taken from the Cloudformation scripts(Pattern1) and converted into the Ansible way to do that.